can/interfaces/canal/canalInterface.py

```diff
@@ -0,0 +1,196 @@
+# this interface is for windows only, otherwise use socketCAN
+
+import logging
+
+from can import BusABC, Message
+from can.interfaces.canal.canal_wrapper import *
+
+bootTimeEpoch = 0
+try:
+    import uptime
+    import datetime
+
+    bootTimeEpoch = (uptime.boottime() - datetime.datetime.utcfromtimestamp(0)).total_seconds()
+except:
+    bootTimeEpoch = 0
+
+# Set up logging
+log = logging.getLogger('can.canal')
+
+
+def format_connection_string(deviceID, baudrate='500'):
+    """setup the string for the device
+
+    config = deviceID + '; ' + baudrate
+    """
+    return "%s; %s" % (deviceID, baudrate)
+
+
+def message_convert_tx(msg):
+    messagetx = CanalMsg()
+
+    length = len(msg.data)
+    messagetx.sizeData = length
+
+    messagetx.id = msg.arbitration_id
+
+    for i in range(length):
+        messagetx.data[i] = msg.data[i]
+
+    messagetx.flags = 0x80000000
+
+    if msg.is_error_frame:
+        messagetx.flags |= IS_ERROR_FRAME
+
+    if msg.is_remote_frame:
+        messagetx.flags |= IS_REMOTE_FRAME
+
+    if msg.id_type:
+        messagetx.flags |= IS_ID_TYPE
+
+    return messagetx
+
+
+def message_convert_rx(messagerx):
+    """convert the message from the CANAL type to pythoncan type"""
+    ID_TYPE = bool(messagerx.flags & IS_ID_TYPE)
+    REMOTE_FRAME = bool(messagerx.flags & IS_REMOTE_FRAME)
+    ERROR_FRAME = bool(messagerx.flags & IS_ERROR_FRAME)
+
+    msgrx = Message(timestamp=messagerx.timestamp / 1000.0,
+                    is_remote_frame=REMOTE_FRAME,
+                    extended_id=ID_TYPE,
+                    is_error_frame=ERROR_FRAME,
+                    arbitration_id=messagerx.id,
+                    dlc=messagerx.sizeData,
+                    data=messagerx.data[:messagerx.sizeData]
+                    )
+
+    return msgrx
+
+
+class CanalBus(BusABC):
+    """Interface to a CANAL Bus.
+
+    Note the interface doesn't implement set_filters, or flush_tx_buffer methods.
+
+    :param str channel:
+        The device's serial number. If not provided, Windows Management Instrumentation
+        will be used to identify the first such device. The *kwarg* `serial` may also be
+        used.
+
+    :param str dll:
+        dll with CANAL API to load
+
+    :param int bitrate:
+        Bitrate of channel in bit/s. Values will be limited to a maximum of 1000 Kb/s.
+        Default is 500 Kbs
+
+    :param str serial (optional)
```

Comment:
Have a look at how this renders in sphinx - I suspect this won't register as a parameter without a trailing `:`

Reply:
I guess that is now solved.